test/Java8andUp/src/org/openj9/test/vmArguments/VmArgumentTests.java

```diff
@@ -325,49 +335,57 @@ public void testCrNocr() {
 		HashMap<String, Integer> posns = checkArguments(actualArguments, new String[] {XCOMPRESSEDREFS, XNOCOMPRESSEDREFS});
 		assertTrue("missing argument: "+crArg, posns.containsKey(crArg));
 		assertTrue("missing argument: "+noCrArg, posns.containsKey(noCrArg));
-		Integer crArgPosn = posns.get(crArg);
-		Integer noCrArgPosn = posns.get(noCrArg);
-		assertTrue(noCrArg+ " should come after "+DJAVA_HOME, crArgPosn < noCrArgPosn);
+		assertTrue(noCrArg+ " should come after "+DJAVA_HOME, posns.get(crArg) < posns.get(noCrArg));
 	}
 
 	/* test options in runnable jar files */
 	
 	public void testJarArguments() {
-		String cmdlineArg = "-DcmdlineArg";
-		String testJar = System.getProperty("org.openj9.test.vmArguments.testJar");
-		URL jarUrl = ClassLoader.getSystemClassLoader().getResource("vmargs.jar");
-		assertNotNull("vmargs.jar not found", jarUrl);
-		testJar = jarUrl.getFile();
-		if (isWindows() && testJar.startsWith("/")) {
-			testJar = testJar.substring(1); /* annoying habit of getResource() to put a / at the front of the path */
+		URL jarUrl = ClassLoader.getSystemClassLoader().getResource(vmargsJarFilename);
+		assertNotNull(vmargsJarFilename+" not found", jarUrl);
+		Path jarPath = null;
+		try {
+			jarPath = Paths.get(jarUrl.toURI()).toFile().toPath();
+			checkJarArgs(jarPath.toString());
+		} catch (URISyntaxException e) {
```

Comment:
Just let this throw.  The test framework will automatically fail the test for you.

Reply:
Please see response below.

Reply:
This exception should never occur in practice given that the test runs and passes before it's merged.  This additional try/catch is extra code making the intent of the test less clear.